CI: post-cluster-api-operator-push-images jobs failing for 89 days

**What steps did you take and what happened:**
See https://github.com/kubernetes-sigs/cluster-api/issues/8784

**What did you expect to happen:**
post-cluster-api-operator-push-images job passing

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
Job was running fine and [passing](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-cluster-api-operator-push-images/1633078125095030784) until March 7th, 2023 and started failing from then. Based on quick look from the older run jobs and latest failing [job](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-cluster-api-operator-push-images/1665467143694585856) run found out "gcc" executable file is not found in the $PATH and thus making `release-staging` Makefile target to fail:

```
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/josharian/intern v1.0.0
# sigs.k8s.io/kustomize/kustomize/v5
/usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exec: "gcc": executable file not found in $PATH
make[2]: Leaving directory '/workspace'
make[2]: *** [Makefile:169: /workspace/hack/tools/bin/kustomize-v5.0.1] Error 2
make[1]: *** [Makefile:329: staging-manifests] Error 2
make[1]: Leaving directory '/workspace'
make: *** [Makefile:421: release-staging] Error 2
ERROR
ERROR: build step 0 "gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20220609-2e4c91eb7e" failed: step exited with non-zero status: 2
```

**To reproduce:**
```
$docker run -it --entrypoint /bin/bash gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20220609-2e4c91eb7e
bash-5.1# go install sigs.k8s.io/kustomize/kustomize/v5@v5.0.1
# sigs.k8s.io/kustomize/kustomize/v5
/usr/local/go/pkg/tool/linux_amd64/link: running gcc failed: exec: "gcc": executable file not found in $PATH
```


/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api-operator/labels?q=area for the list of labels]

--- a/Makefile
+++ b/Makefile
@@ -166,7 +166,7 @@ yq: $(YQ) ## Build a local copy of yq.
 kpromo: $(KPROMO) ## Build a local copy of kpromo.
 
 $(KUSTOMIZE): ## Build kustomize from tools folder.
-	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) sigs.k8s.io/kustomize/kustomize/v5 $(KUSTOMIZE_BIN) $(KUSTOMIZE_VER)
+	CGO_ENABLED=0 GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) sigs.k8s.io/kustomize/kustomize/v5 $(KUSTOMIZE_BIN) $(KUSTOMIZE_VER)
 
 $(GO_APIDIFF): ## Build go-apidiff from tools folder.
 	GOBIN=$(TOOLS_BIN_DIR) $(GO_INSTALL) github.com/joelanford/go-apidiff $(GO_APIDIFF_BIN) $(GO_APIDIFF_VER)

